src/Microsoft.Health.Fhir.Shared.Core/Features/Resources/Create/CreateResourceHandler.cs

```diff
@@ -70,7 +76,16 @@ public async Task<UpsertResourceResponse> Handle(CreateResourceRequest message,
 
             resource.VersionId = result.Wrapper.Version;
 
-            return new UpsertResourceResponse(new SaveOutcome(new RawResourceElement(result.Wrapper), SaveOutcomeType.Created));
+            var response = new UpsertResourceResponse(new SaveOutcome(new RawResourceElement(result.Wrapper), SaveOutcomeType.Created));
+
+            if (message.Resource.InstanceType.Equals(KnownResourceTypes.SearchParameter, StringComparison.Ordinal))
+            {
+                // Once the SearchParameter resource is committed to the data store, we can update the in
+                // memory SearchParameterDefinitionManager, and persist the status to the data store
+                await _searchParameterUtitliies.AddSearchParameterAsync(message.Resource.Instance);
```

Comment:
>await _searchParameterUtitliies.AddSearchParameterAsync(message.Resource.Instance); [](start = 16, length = 83)

So if this one fail (let's say on status update for search parameter, since it's a work with external resource), what exactly we gonna return to user? I'm sure we catch exception in filter, but would it be some generic thing or something more specific.
What kind of actions user need to perform to recover from such error? Does it need to just run reindex?

Reply:
Good point!  After checking, depending on the exception it will fail with a 500 and give a full stack trace with no formatted exception at all.  Updated with better exception handling in the SearchParameterUtilities.

Reply:
This is when I'd separate with INotification, could be a separate PR, but something like `ResourceCreatedNotification : INotification`. These will be useful and probably required for things like subscriptions.

Alternatively, this hook is also available with `IPipelineBehavior<CreateResourceRequest, UpsertResourceResponse>>` or `RequestPostProcessorBehavior<CreateResourceRequest, UpsertResourceResponse>`

Reply:
This is good info @brendankowitz , I added a story to review and possibly reduce the coupling using mediatr to hook up this behavior.  AB#78200